db/db_impl_open.cc

```diff
@@ -412,6 +432,16 @@ Status DBImpl::Recover(
     default_cf_handle_ = new ColumnFamilyHandleImpl(
         versions_->GetColumnFamilySet()->GetDefault(), this, &mutex_);
     default_cf_internal_stats_ = default_cf_handle_->cfd()->internal_stats();
+    // When recovering from a DB which already contains persistent stats CF,
+    // the CF is already created in VersionSet::ApplyOneVersionEdit, but
+    // column family handle was not. Need to explicitly create handle here.
+    if (mutable_db_options_.stats_persist_period_sec != 0 &&
+        persist_stats_cf_handle_ == nullptr) {
+      persist_stats_cf_handle_ = new ColumnFamilyHandleImpl(
+          versions_->GetColumnFamilySet()->GetColumnFamily(
+              kPersistentStatsColumnFamilyName),
+          this, &mutex_);
+    }
     single_column_family_mode_ =
```

Comment:
`single_column_family_mode_ = versions_->GetColumnFamilySet()->NumberOfColumnFamilies() == 1;` is no longer true if there is persistent stats column family.

Reply:
yes, this needs to be addressed